sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/FileWriteResult.java

```diff
@@ -15,7 +15,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.beam.io.iceberg;
+package org.apache.beam.sdk.io.iceberg;
 
```

Comment:
This class looks like it's used for actual data which may cause backwards compatibility headaches in the future. Rather than a custom FileWriteResult.FileWriteResultCoder.class, could we declare it as a AutoSchema and use the SchemaCoder?

Reply:
Can't speak to this design decision, but it seems to be doing a lot of custom encoding with the help of some Apache Iceberg utils. I don't know if SchemaCoder would capture this complexity

Reply:
If there's complexity here that's all the more reason to not bake this into the implementation and tie our hands with update compatibility, right?

Reply:
We decided for now to encode a "version number" along with the data so that in the future we can fork if needed